hack/shared-cluster.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# shared-cluster.sh is used to provide a repeatable production cluster create script
+
+LOCATION="westcentralus"
+CLUSTER="shared-cluster"
+RESOURCEGROUP="shared-cluster"
+CLUSTERRESOURCEGROUP="aro-shared-cluster"
+
+usage() {
+	echo -e "usage: ${0} <create|delete>"
+	exit 1
+}
+
+if [ "$#" -ne 1 ]; then
+	usage
+fi
+
+if [[ $1 == create ]]; then
+    echo "creating resource group and network"
+    az group create --name $RESOURCEGROUP --location $LOCATION
+
+    az network vnet create \
+      --resource-group $RESOURCEGROUP \
+      --name aro-vnet \
+      --address-prefixes 10.0.0.0/22
+
+    az network vnet subnet create \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --name master-subnet \
+      --address-prefixes 10.0.0.0/23 \
+      --service-endpoints Microsoft.ContainerRegistry
+
+    az network vnet subnet create \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --name worker-subnet \
+      --address-prefixes 10.0.2.0/23 \
+      --service-endpoints Microsoft.ContainerRegistry
+
+    az network vnet subnet update \
+      --name master-subnet \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --disable-private-link-service-network-policies true
+
+    echo "resource group and vnet/subnets created..."
+    echo "creating cluster..."
+    az aro create \
```

Comment:
Do we need to unset RP_MODE for consistency here, or have a check that it's unset?

Reply:
RP_MODE should not be set. the instructions for creating/deleting cluster do not include sourcing `env` file. Instructions are here: https://github.com/Azure/ARO-RP/pull/2187/files#diff-a7eb7df5033da5999c603571053ea9a6811ec9aac786279fb7e5ba61935c182bR55

Reply:
this will also be the cloudshell az cli, which will not have development aro extension and will not honor `RP_MODE`